### PR TITLE
Feature1

### PR DESCRIPTION
**Pull request by Vlad-Andreychik**

### CODE REVIEW

File: clients/http_client.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import requests
+
+
+class HTTPClient:
+    """
+    HTTP client for jobs.tut.by
```
Comment (akopylov89) [inline clients/http_client.py]: it is not for jobs.tut.by, it could be used for all requests

File: fixtures/rabota_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import pytest
```
Comment (akopylov89) [inline fixtures/rabota_fixtures.py:6]: Module docstring

File: clients/http_client.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import requests
```
Comment (akopylov89) [inline clients/http_client.py:4]: Module docstring

File: models/parser_jobs_tut_by.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    def get_description_for_first_vacancy():
+        """
+        Returns description for first vacancy
+        """
+        headers = {
+            'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/webkit-version (KHTML, like Gecko) '
+                          'Silk / browser - version like Chrome / chrome - version Safari / webkit - version'}
+        url = 'https://rabota.by/search/' \
+              'vacancy?L_is_autosearch=false&area=16&clusters=true&enable_snippets=true&text=python&page=0'
+        response = HTTPClient.get(url, headers=headers).text
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: It is incorrect. RabotaByParser should receive some page as init argument and parse it. Do not use HTTPClient inside

File: models/parser_jobs_tut_by.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+        lxml_text = RabotaByParser.get_lxml(response)
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: RabotaByParser.get_lxml(response) should be self.get_lxml(response)

File: models/parser_jobs_tut_by.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        list_id = RabotaByParser.get_list_of_vacancy_id(lxml_text)
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: the same

File: models/parser_jobs_tut_by.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        first_vacancy_id = list_id[0]
+        response_for_first_vacancy = HTTPClient.get(f'https://rabota.by/vacancy/{first_vacancy_id}?query=python',
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: incorrect

File: models/parser_jobs_tut_by.py [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+                                                    headers=headers).text
+        description = RabotaByParser.find_vacancy_description(response_for_first_vacancy)
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: incorrect

File: models/parser_jobs_tut_by.py [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+        Returns average amount of occurrences for words python, linux, flask
+        """
+        headers = {
+            'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/webkit-version (KHTML, like Gecko) '
+                          'Silk / browser - version like Chrome / chrome - version Safari / webkit - version'}
+        url_for_page = 'https://rabota.by/search/' \
+                       'vacancy?L_is_autosearch=false&area=16&clusters=true&enable_snippets=true&text=python&page='
+        url = 'https://rabota.by/search/' \
+              'vacancy?L_is_autosearch=false&area=16&clusters=true&enable_snippets=true&text=python&page=0'
+        response = HTTPClient.get(url, headers=headers).text
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: incorrect

File: models/parser_jobs_tut_by.py [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+            count_flask = RabotaByParser.count_word('flask', description)
+            dictionary[f'{number} vacancy'] = {'python': count_python, 'linux': count_linux, 'flask': count_flask}
+            python += count_python
+            linux += count_linux
+            flask += count_flask
+            number += 1
+        avg_python = python // len(all_urls_list)
+        avg_linux = linux // len(all_urls_list)
+        avg_flask = flask // len(all_urls_list)
+        return avg_python, avg_linux, avg_flask
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: It is awful method!!! Create a method which receives a word as argument and counts it. ONLY ONE WORD!

File: models/parser_jobs_tut_by.py [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+    def amount_vacancies_on_one_page():
+        """
+        Returns amount of vacancies founded from query on one page
+        """
+        headers = {
+            'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/webkit-version (KHTML, like Gecko) '
+                          'Silk / browser - version like Chrome / chrome - version Safari / webkit - version'}
+        url = 'https://rabota.by/search/' \
+              'vacancy?L_is_autosearch=false&area=16&clusters=true&enable_snippets=true&text=python&page=0'
+        response = HTTPClient.get(url, headers=headers)
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: incorrect

File: models/parser_jobs_tut_by.py [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+    def amount_of_vacancies_founded(query):
+        """
+        Returns amount of vacancies founded from query
+        """
+        headers = {
+            'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/webkit-version (KHTML, like Gecko) '
+                          'Silk / browser - version like Chrome / chrome - version Safari / webkit - version'}
+        url = f'https://rabota.by/search/' \
+              f'vacancy?L_is_autosearch=false&area=16&clusters=true&enable_snippets=true&text={query}&page=0'
+        response = HTTPClient.get(url, headers=headers).text
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: incorrect

File: models/parser_jobs_tut_by.py [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+        lxml_text = RabotaByParser.get_lxml(response)
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: incorrect

File: models/parser_jobs_tut_by.py [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+        Returns amount of vacancies founded from query on one page
+        """
+        headers = {
+            'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/webkit-version (KHTML, like Gecko) '
+                          'Silk / browser - version like Chrome / chrome - version Safari / webkit - version'}
+        url = 'https://rabota.by/search/' \
+              'vacancy?L_is_autosearch=false&area=16&clusters=true&enable_snippets=true&text=python&page=0'
+        response = HTTPClient.get(url, headers=headers)
+        lxml_text = RabotaByParser.get_lxml(response.text)
+        list_id = RabotaByParser.get_list_of_vacancy_id(lxml_text)
```
Comment (akopylov89) [inline models/parser_jobs_tut_by.py]: incorrect

File: tests/rabota_by_tests.py [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+    assert get_response.status_code == 200, 'Connection error'
+
+
+def test_shotgun_not_found(get_query_shotgun):
+    assert 'По запросу «shotgun» ничего не найдено' in get_query_shotgun.text, 'Query give responses'
+
+
+def test_occurrence_of_the_words(get_description_for_first_vacancy, avg_occurrences_for_word_python,
+                                 avg_occurrences_for_word_linux, avg_occurrences_for_word_flask):
+    description = get_description_for_first_vacancy
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: useless

File: tests/rabota_by_tests.py [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+
+from models.parser_jobs_tut_by import RabotaByParser
+
+
+def test_connection(get_response):
+    assert get_response.status_code == 200, 'Connection error'
+
+
+def test_shotgun_not_found(get_query_shotgun):
+    assert 'По запросу «shotgun» ничего не найдено' in get_query_shotgun.text, 'Query give responses'
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: How can you understand what has happened from 'Query give responses' assert message&

File: tests/rabota_by_tests.py [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+
+
+def test_occurrence_of_the_words(get_description_for_first_vacancy, avg_occurrences_for_word_python,
+                                 avg_occurrences_for_word_linux, avg_occurrences_for_word_flask):
+    description = get_description_for_first_vacancy
+    count_python = RabotaByParser.count_word('python', description)
+    count_linux = RabotaByParser.count_word('linux', description)
+    count_flask = RabotaByParser.count_word('flask', description)
+    assert count_python in range(avg_occurrences_for_word_python[0] - 1, avg_occurrences_for_word_python[0] + 2)
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: assert message

File: tests/rabota_by_tests.py [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+def test_accordance_amount_pages_for_amount_of_vacancies(amount_of_vacancies_founded_for_python,
+                                                         amount_vacancies_on_one_page,
+                                                         get_response):
+    actual_number_of_pages = RabotaByParser.amount_of_pages(get_response.text)
+    estimated_number_of_pages = math.ceil(amount_of_vacancies_founded_for_python / amount_vacancies_on_one_page)
+    assert estimated_number_of_pages == int(actual_number_of_pages), 'Incorrect amount of pages'
+
+
+def test_wrong_url(get_response_from_wrong_url):
+    assert get_response_from_wrong_url.status_code == 404, 'Correct url entered'
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: 'Correct url entered'?

File: tests/rabota_by_tests.py [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+def test_amount_vacancies_on_one_page(amount_vacancies_on_one_page):
+    assert amount_vacancies_on_one_page == 50, 'Incorrect amount of vacancies on one page'
+
+
+def test_accordance_amount_pages_for_amount_of_vacancies(amount_of_vacancies_founded_for_python,
+                                                         amount_vacancies_on_one_page,
+                                                         get_response):
+    actual_number_of_pages = RabotaByParser.amount_of_pages(get_response.text)
+    estimated_number_of_pages = math.ceil(amount_of_vacancies_founded_for_python / amount_vacancies_on_one_page)
+    assert estimated_number_of_pages == int(actual_number_of_pages), 'Incorrect amount of pages'
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: don't have this logic in tests. it should be returned from fixture

File: clients/http_client.py (hunk 1 above)
Comment (Vlad-Andreychik) [reply clients/http_client.py]: Fixed

File: fixtures/rabota_fixtures.py (hunk 2 above)
Comment (Vlad-Andreychik) [reply fixtures/rabota_fixtures.py:6]: Done

File: clients/http_client.py (hunk 3 above)
Comment (Vlad-Andreychik) [reply clients/http_client.py:4]: Done

File: tests/rabota_by_tests.py (hunk 19 above)
Comment (Vlad-Andreychik) [reply tests/rabota_by_tests.py]: Fixed

File: tests/rabota_by_tests.py (hunk 18 above)
Comment (Vlad-Andreychik) [reply tests/rabota_by_tests.py]: Fixed

File: tests/rabota_by_tests.py (hunk 16 above)
Comment (Vlad-Andreychik) [reply tests/rabota_by_tests.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 14 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 13 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 12 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 11 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 9 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 8 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 7 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 6 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 5 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

File: models/parser_jobs_tut_by.py (hunk 4 above)
Comment (Vlad-Andreychik) [reply models/parser_jobs_tut_by.py]: Fixed

**Pull request closed without merging**